x/incentives/keeper/distribute.go

```diff
@@ -196,26 +199,28 @@ func (d *distributionInfo) addLockRewards(owner string, rewards sdk.Coins) error
 // doDistributionSends utilizes provided distributionInfo to send coins from the module account to various recipients.
 func (k Keeper) doDistributionSends(ctx sdk.Context, distrs *distributionInfo) error {
 	numIDs := len(distrs.idToDecodedAddr)
-	ctx.Logger().Debug(fmt.Sprintf("Beginning distribution to %d users", numIDs))
-	err := k.bk.SendCoinsFromModuleToManyAccounts(
-		ctx,
-		types.ModuleName,
-		distrs.idToDecodedAddr,
-		distrs.idToDistrCoins)
-	if err != nil {
-		return err
+	if numIDs > 0 {
+		ctx.Logger().Debug(fmt.Sprintf("Beginning distribution to %d users", numIDs))
+		err := k.bk.SendCoinsFromModuleToManyAccounts(
+			ctx,
+			types.ModuleName,
+			distrs.idToDecodedAddr,
+			distrs.idToDistrCoins)
+		if err != nil {
+			return err
+		}
+		ctx.Logger().Debug("Finished sending, now creating liquidity add events")
+		for id := 0; id < numIDs; id++ {
```

Comment:
Out of curiosity, why would we emit an event for each ID?

Reply:
it's for tokens sent from moduletoUserAccounts, therefore at every successful transfer an event is emitted